components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+      tags:
+        - me
+
+  /me/push-auth/devices:
+    post:
+      description: |
+        This API is used to receive device information from the mobile app and complete the add account flow.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_create
+      summary: |
+        Register a device for push-based authentication.
+      requestBody:
+        description: 'Device details sent by mobile application'
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RegistrationRequestDTO'
+      responses:
+        201:
+          description: Added new device
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StatusDTO"
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        409:
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - me
+
+    get:
+      description: |
+        This API is used to get a list of the registered devices.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns registered devices list of the user.
+      responses:
+        200:
+          description: All availabe devices of the user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AllDevices'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        404:
```

Comment:
Can this api result in a 404 response? or when no devices are registered does it send back a empty array with a 200 response?

Reply:
Should be removed

Reply:
Removed